tests/test_helpers.py

```diff
@@ -234,18 +234,32 @@ def test_name_with_import_error(self, modules_tmp_path):
 
 
 class TestStreaming:
-    def test_streaming_with_context(self, app, client):
-        @app.route("/")
-        def index():
+    def test_stream_with_context_fails_with_async_route(self):
+        import gc
+
+        import pytest
+
+        import flask
+
+        app = flask.Flask(__name__)
+
+        @app.route("/stream")
+        async def stream():
+            @flask.stream_with_context
             def generate():
-                yield "Hello "
-                yield flask.request.args["name"]
-                yield "!"
+                yield "hello"
 
-            return flask.Response(flask.stream_with_context(generate()))
+            return flask.Response(generate())
 
-        rv = client.get("/?name=World")
-        assert rv.data == b"Hello World!"
+        with app.test_request_context("/stream"):
```

Comment:
Why are you using `with test_request_context` rather than making a request with the client like the previous test?